### ISSUE TITLE
Automated Breast Ultrasound Lesions Detection Using Convolutional Neural Networks

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#156]
## Abstract & Introduction
* deep learning을 이용한 breast 초음파 상의 lesion detection 방법 제안.
* 3가지 methods 제시 (Patch-based LeNet, U-Net, transfer learning approach)
* 비교 대상으로 기존 방법도 해봄 (Radial Gradient Index, Multifractal Filtering, Rule-based Region Ranking, Deformable Part Models)
* digital mammography (DM) -> ultrasound imaging -> computer aided diagnosis (CAD)
* CAD의 도전과제들
  * the ability to locate the lesion
  * the lack of a public standard dataset

## Related Work
* Radial Gradient Index, Multifractal Filtering, Rule-based Region Ranking, Deformable Part Models에 대한 설명해줌.
* Deep learning for Breast Imaging
  * Patch-based CNNs approach
  * Fully convolutional approach
  * Transfer learning approach

## Datasets
* Dataset A와 Dataset B가 있는데 A는 2001년에 수집, B는 2012년에 수집되어 둘의 화질과 noise 정도가 매우 다름.
* 각 병변 종류에 따른 Dataset 사진 예시. (위가 Dataset A)
  <image src="https://user-images.githubusercontent.com/67627471/133928366-2a3d9ffb-92a5-4d9e-b840-f3bca1ef324e.png" width="50%" height="50%" />

## Methodology
* CNN에서 conv layers --> extract local features from a set of learnable filters
* pooling layers --> merge neighbouring patterns

#### Patch-based LeNet
* LeNet 전체적인 구조.
  <image src="https://user-images.githubusercontent.com/67627471/133928392-325fb4c5-f21d-4e6d-a219-ed1effc8cb05.png" width="60%" height="60%" />
* 초음파 사진이 grayscale이고 lesion 크기가 상대적으로 작아서 LeNet이 two-class classification 문제 (병변있는 곳/없는 곳) 해결에 적합함.
* breast ultrasound images --> split into 28 x 28 patches
* Root Mean Square Propagation (RMSProp)로 학습시킴.
* lr=0.01 & 60epochs & dr=0.33

#### U-Net
* encoder-decoder based CNN with skip connections.
* original image와 gt training label로 학습.
* Adam optimizer로 학습 진행.
* lr = 0.0001 & 300 epochs

#### Trasnfer Learning
* 본 논문에서는 fully convolutional network인 FCN-AlexNet for semantic segmentation을 기본 뼈대로 사용함.
* 위 네트워크는 원래 ImageNet dataset에서 1000개의 다른 object를 classification하는 문제에 쓰였음.
* 여기도 원본 이미지와 label 이미지로 학습.
* lr = 0.001 & 60 epochs & dr = 0.33

#### U-Net과 FCN-AlexNet 학습시킬 때 사용한 인풋 이미지들 예시.
  <image src="https://user-images.githubusercontent.com/67627471/133929113-3d140809-b797-4678-bb8f-9efd5dd9584a.png" width="40%" height="40%" />

## Results
* 마지막줄은 lesion이 너무 작아서 모든 모델이 전부 detection에 실패했다고 함.
  <image src="https://user-images.githubusercontent.com/67627471/133929373-71a75ed4-4c23-436b-baee-fc1f5c706893.png" width="70%" height="70%" />
  <image src="https://user-images.githubusercontent.com/67627471/133929414-32e42a69-e020-4433-8c65-773e466c4966.png" width="40%" height="40%" />  <image src="https://user-images.githubusercontent.com/67627471/133929428-f8d71527-c83c-4217-b7e9-e24b12692b20.png" width="50%" height="50%" />
